This patch (inspired by Debian) allows ghc-haddock-library to use our
ghc-attoparsec package instead of using a bundled version.

--- a/haddock-library.cabal	2018-09-01 01:22:18.676855884 -0400
+++ b/haddock-library.cabal	2018-09-01 01:25:10.501150260 -0400
@@ -10,7 +10,6 @@
                       itself, see the ‘haddock’ package.
 license:              BSD3
 license-files:        LICENSE
-                      vendor/attoparsec-0.13.1.0/LICENSE
 maintainer:           Alex Biehl <alexbiehl@gmail.com>, Simon Hengel <sol@typeful.net>, Mateusz Kowalczyk <fuuzetsu@fuuzetsu.co.uk>
 homepage:             http://www.haskell.org/haddock/
 bug-reports:          https://github.com/haskell/haddock/issues
@@ -28,7 +27,6 @@
     , containers   >= 0.4.2.1 && < 0.6
     , transformers >= 0.3.0   && < 0.6
 
-  -- internal sub-lib
   build-depends:        attoparsec
 
   hs-source-dirs:       src
@@ -49,42 +47,6 @@
   if impl(ghc >= 8.0)
     ghc-options: -Wcompat -Wnoncanonical-monad-instances -Wnoncanonical-monadfail-instances
 
-library attoparsec
-  default-language:     Haskell2010
-
-  build-depends:
-      base         >= 4.5     && < 4.12
-    , bytestring   >= 0.9.2.1 && < 0.11
-    , deepseq      >= 1.3     && < 1.5
-
-  hs-source-dirs:       vendor/attoparsec-0.13.1.0
-
-  -- NB: haddock-library needs only small part of lib:attoparsec
-  --     internally, so we only bundle that subset here
-  exposed-modules:
-    Data.Attoparsec.ByteString
-    Data.Attoparsec.ByteString.Char8
-    Data.Attoparsec.Combinator
-
-  other-modules:
-    Data.Attoparsec
-    Data.Attoparsec.ByteString.Buffer
-    Data.Attoparsec.ByteString.FastSet
-    Data.Attoparsec.ByteString.Internal
-    Data.Attoparsec.Internal
-    Data.Attoparsec.Internal.Fhthagn
-    Data.Attoparsec.Internal.Types
-    Data.Attoparsec.Number
-
-  ghc-options:          -funbox-strict-fields -Wall -fwarn-tabs -O2
-
-  ghc-options: -Wall
-  if impl(ghc >= 8.0)
-    ghc-options: -Wcompat -Wnoncanonical-monad-instances -Wnoncanonical-monadfail-instances
-  else
-    build-depends: semigroups ^>= 0.18.3, fail ^>= 4.9.0.0
-
-
 test-suite spec
   type:             exitcode-stdio-1.0
   default-language: Haskell2010
@@ -115,11 +77,10 @@
     , hspec         ^>= 2.4.4
     , QuickCheck    ^>= 2.11
 
-  -- internal sub-lib
   build-depends: attoparsec
 
   -- Versions for the dependencies below are transitively pinned by
-  -- dependency on haddock-library:lib:attoparsec
+  -- dependency on attoparsec
   build-depends:
       base
     , bytestring
@@ -146,7 +107,7 @@
     haddock-library
 
   -- Versions for the dependencies below are transitively pinned by
-  -- dependency on haddock-library:lib:attoparsec
+  -- dependency on attoparsec
   build-depends:
       base
 
